app/ts/background/simulationModeHanders.ts

```diff
@@ -32,35 +32,32 @@ export async function getTransactionReceipt(ethereumClientService: EthereumClien
 	return { result: EthTransactionReceiptResponse.serialize(await getSimulatedTransactionReceipt(ethereumClientService, simulationState, request.params[0])) }
 }
 
-function getFromField(websiteTabConnections: WebsiteTabConnections, simulationMode: boolean, transactionFrom: bigint | undefined, getActiveAddressForDomain: (websiteAccess: WebsiteAccessArray, websiteOrigin: string, settings: Settings) => bigint | undefined, socket: WebsiteSocket, settings: Settings) {
+function getFromField(websiteTabConnections: WebsiteTabConnections, simulationMode: boolean, transactionFrom: bigint | undefined, activeAddress: bigint | undefined, socket: WebsiteSocket) {
 	if (simulationMode && transactionFrom !== undefined) {
 		return transactionFrom // use `from` field directly from the dapp if we are in simulation mode and its available
 	} else {
 		const connection = getConnectionDetails(websiteTabConnections, socket)
 		if (connection === undefined) throw new Error('Not connected')
-
-		const from = getActiveAddressForDomain(settings.websiteAccess, connection.websiteOrigin, settings)
-		if (from === undefined) throw new Error('Access to active address is denied')
-		return from
+		if (activeAddress === undefined) throw new Error('Access to active address is denied')
+		return activeAddress
 	}
 }
 
 export async function sendTransaction(
 	websiteTabConnections: WebsiteTabConnections,
-	getActiveAddressForDomain: (websiteAccess: WebsiteAccessArray, websiteOrigin: string, settings: Settings) => bigint | undefined,
+	activeAddress: bigint | undefined,
```

Comment:
Same comment as above (and elsewhere), does optionality better express what you want, or do you expect an explicitly undefined `activeAddress`?